tests/tests/operation_tests.cpp

```diff
@@ -556,6 +556,127 @@ BOOST_AUTO_TEST_CASE( more_call_order_update_test_after_hardfork_583 )
+      asset_id_type core_id = core.id;
+
+      transfer(committee_account, sam_id, asset(10000000));
+      transfer(committee_account, alice_id, asset(10000000));
+      transfer(committee_account, bob_id, asset(10000000));
+      update_feed_producers( bitusd, {sam.id} );
+
+      price_feed current_feed; current_feed.settlement_price = bitusd.amount( 100 ) / core.amount(100);
+      current_feed.maintenance_collateral_ratio = 1750; // need to set this explicitly, testnet has a different default
+      current_feed.maximum_short_squeeze_ratio = 1100; // need to set this explicitly, testnet has a different default
+      publish_feed( bitusd, sam, current_feed );
+
+      FC_ASSERT( bitusd.bitasset_data(db).current_feed.settlement_price == current_feed.settlement_price );
+
+      BOOST_TEST_MESSAGE( "alice tries to borrow using 4x collateral at 1:1 price with target_cr set, "
+                          "will fail before hard fork time" );
+      GRAPHENE_REQUIRE_THROW( borrow( alice, bitusd.amount(100000), core.amount(400000), 0 ), fc::assert_exception );
+      GRAPHENE_REQUIRE_THROW( borrow( alice, bitusd.amount(100000), core.amount(400000), 1 ), fc::assert_exception );
+      GRAPHENE_REQUIRE_THROW( borrow( alice, bitusd.amount(100000), core.amount(400000), 1749 ), fc::assert_exception );
+      GRAPHENE_REQUIRE_THROW( borrow( alice, bitusd.amount(100000), core.amount(400000), 1750 ), fc::assert_exception );
+      GRAPHENE_REQUIRE_THROW( borrow( alice, bitusd.amount(100000), core.amount(400000), 1751 ), fc::assert_exception );
+      GRAPHENE_REQUIRE_THROW( borrow( alice, bitusd.amount(100000), core.amount(400000), 65535 ), fc::assert_exception );
+
+      auto call_update_proposal = [&]( const account_object& proposer,
+                                       const account_object& updater,
+                                       const asset& delta_collateral,
+                                       const asset& delta_debt,
+                                       const optional<uint16_t> target_cr )
+      {
+         call_order_update_operation op;
+         op.funding_account = updater.id;
+         op.delta_collateral = delta_collateral;
+         op.delta_debt = delta_debt;
+         op.extensions.value.target_collateral_ratio = target_cr;
+
+         const auto& curfees = *db.get_global_properties().parameters.current_fees;
+         const auto& proposal_create_fees = curfees.get<proposal_create_operation>();
+         proposal_create_operation prop;
+         prop.fee_paying_account = proposer.id;
+         prop.proposed_ops.emplace_back( op );
+         prop.expiration_time =  db.head_block_time() + fc::days(1);
+         prop.fee = asset( proposal_create_fees.fee + proposal_create_fees.price_per_kbyte );
+
+         signed_transaction tx;
+         tx.operations.push_back( prop );
+         db.current_fee_schedule().set_fee( tx.operations.back() );
+         set_expiration( db, tx );
+         db.push_transaction(trx, ~0);
```

Comment:
You're pushing the wrong transaction here... :-)

Reply:
This kind of bug is hard to detect, especially when the code is in a lambda, changing `[&]` to `[this]` here doesn't help. :-( 

What's best practice?